src/thriveopedia/fossilisation/FossilisationDialog.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System.Linq;
+using System.Text.RegularExpressions;
+using Godot;
+
+/// <summary>
+///   Dialog for fossilising (saving) a given species.
+/// </summary>
+public class FossilisationDialog : CustomDialog
+{
+    [Export]
+    public NodePath NameEditPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    [Export]
+    public NodePath FossiliseButtonPath = null!;
+
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath OverwriteNameConfirmationDialogPath = null!;
+
+    private LineEdit speciesNameEdit = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private Button fossiliseButton = null!;
+    private CustomConfirmationDialog overwriteNameConfirmationDialog = null!;
+
+    /// <summary>
+    ///   The species currently open in the dialog.
+    /// </summary>
+    private Species selectedSpecies = null!;
+
+    /// <summary>
+    ///   True when one of our (name related) Controls is hovered. This needs to be known to know if a click happened
+    ///   outside the name editing controls, for detecting when the name needs to be validated.
```

Comment:
Can a lose focus callback from the name box be used instead? As that would work for controller input as well (and keyboard navigation).

Reply:
I believe I copied this wholesale from EditorComponentBottomLeftButtons, so if this is changed it's probably best to change that too.

Reply:
It does seem like it, so that would be good to update as well. I just searched for "OnClickedOffName" and it seems like you didn't attach a callback to this method, so it would seem to be that this cannot currently function correctly.

Reply:
Having looked into this, I think I can get away with removing it. The editor blocks any other input until the name is valid which is why this is needed there, but I don't need that here as I'm just disabling the fossilisation button when the name is invalid and the player is free to perform other actions.

Reply:
That sounds fine.